addons/mail/models/mail_message.py

```diff
@@ -1171,10 +1171,17 @@ def _notify_message_notification_update(self):
                     continue
                 else:
                     messages |= message
+        messages_per_partner = defaultdict(lambda: self.env['mail.message'])
+        for message in messages:
+            partner = self.env.user.partner_id
+            author = message.author_id
```

Comment:
Unnecessary variable: you may use directly message_author_id in the sub-statements